axiom-eth/src/rlp/mod.rs

```diff
@@ -587,9 +768,10 @@ impl<'range, F: ScalarField> RlpChip<'range, F> {
                 len_trace: len_rlc,
                 field_trace: field_rlc,
             });
+            cml_max_len += field_witness.max_field_len;
         }
 
-        rlc.load_rlc_cache((ctx_gate, ctx_rlc), self.gate(), bit_length(rlp_array.len() as u64));
+        rlc.load_rlc_cache((ctx_gate, ctx_rlc), self.gate(), bit_length(cml_max_len as u64));
```

Comment:
This is an optimization right? 

In fact after looking into it, we only use `rlc_cache` in `constrain_rlc_concat` so it seems like taking the `max` of all `field_witness.max_field_len` would work?

Reply:
This is not an optimization -- it fixes a bug.  If rlp_array.len() is (much) shorter than the maximum length, it won't load enough powers of gamma, and the assertion at rlc.rs:334 will fail.
https://github.com/axiom-crypto/axiom-eth/blob/87a8747bca788885be5e5fd8507a6ba8ee1657ce/axiom-eth/src/rlp/rlc.rs#L334

Yes, I think the max of all field_witness.max_field_len would work.

Reply:
Ah, we actually resize so rlp_array is always the max length. The max_length field is actually redundant.

We'll make this clearer by using a VarLenBytes struct soon.